snuba/utils/streams/backends/kafka.py

```diff
@@ -695,6 +698,17 @@ def get_default_kafka_configuration(
 
     if override_params:
         broker_config.update(override_params)
+
+    broker_config.update(
+        {
+            "log_level": pylog_to_syslog_level(getattr(logging, settings.LOG_LEVEL)),
+            "log.queue": "true",
```

Comment:
https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
>Disable spontaneous log_cb from internal librdkafka threads, instead enqueue log messages on queue set with rd_kafka_set_log_queue() and serve log callbacks or events through the standard poll APIs. NOTE: Log messages will linger in a temporary queue until the log queue has been set.

Reply:
How does it help us here? Why are we setting it now?

Reply:
My guess was this directly pushing things to `stdout` instead of collaborating with Python's logger. We can try removing this and see how broken this is.

Reply:
this is not only for the bootstrap. This code is used to consume and produce in production. 

There are a few issues.
`"log_level": pylog_to_syslog_level(getattr(logging, settings.LOG_LEVEL)),` 
The log level may not be set through settings in the consumer. It can be overridden on the CLI: https://github.com/getsentry/snuba/blob/master/snuba/cli/consumer.py#L69
So this change will ignore the case where it is actually overridden, which is not desirable.

`"log.queue": "true",`
Did you test logging actually happens through python logger now?
In theory the python logger should be used by default, but I am no sure it actually works in our version. https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#kafka-client-configuration
Moreover that would collect logs only during `poll` and `flush` which I am not sure cover all the use cases. I can look more in depth on this later.

Reply:
>this is not only for the bootstrap. This code is used to consume and produce in production.

I'm well aware of that, this was intentional. Sorry that the title does not cover this fully as this part came later in the game when I realized I was still getting some noise.

>The log level may not be set through settings in the consumer. It can be overridden on the CLI:

Great point, I'll make sure this is addressed.

>"log.queue": "true",
>Did you test logging actually happens through python logger now?

I haven't really and there seems to be some concerns around this setting so I'm okay leaving it alone if it doesn't really change the outcome I'm looking for.